config/ebcdic500_latin1.ttbl

```diff
@@ -0,0 +1 @@
+000102039C09867F978D8E0B0C0D0E0F101112139D8508871819928F1C1D1E1F80818283840A171B88898A8B8C050607909116939495960498999A9B14159E1A20A0E2E4E0E1E3E5E7F15B2E3C282B2126E9EAEBE8EDEEEFECDF5D242A293B5E2D2FC2C4C0C1C3C5C7D1A62C255F3E3FF8C9CACBC8CDCECFCC603A2340273D22D8616263646566676869ABBBF0FDFEB1B06A6B6C6D6E6F707172AABAE6B8C6A4B57E737475767778797AA1BFD0DDDEAEA2A3A5B7A9A7B6BCBDBEAC7CAFA8B4D77B414243444546474849ADF4F6F2F3F57D4A4B4C4D4E4F505152B9FBFCF9FAFF5CF7535455565758595AB2D4D6D2D3D530313233343536373839B3DBDCD9DA9F
```

Comment:
as this is all two-byte hex as string please adjust it to use 513 bytes, switching to the translation part at first line break (it is just much easier to match that way) and move the gcos translation table to a file on the way

Reply:
Should I keep my code that allows any trailing blanks after the table ?

Reply:
> Should I keep my code that allows any trailing blanks after the table ?

Let's adjust that to just skip any space (also in between) and read until file end. Decide for yourself if the match needs to be "complete" (256 translations) or if it makes more reason to have "missing entries are filled up using the last one" (I don't know).

In any case docs/gnucobol.texi should have a note on this format.

Reply:
... and possibly skip the line if it starts with a comment `#` or is empty?

Reply:
Alright, I changed to a more flexible format, allowing comment lines starting with # and allowing blank spaces everywhere (except between the two digits of a single byte).

And now I'm just realizing the translations are not necessarily symmetric (specifically when using substitution characters).
What do you think if the table file contained not 256 bytes but 512, the first half for EBCDIC to ASCII and the other half for ASCII to EBCDIC ? Also, to ease input of symmetric translations, we could allow for files containing only 256 bytes and in this case build the complement automatically (as I'm already doing right now).

Reply:
Make this depending on "4 data row" or "2 data rows" and that's fine.

Reply:
Sorry I'm not sure what you mean about that.

For info, currently the format allows to write the table like this :
```
# EBCDIC 500 to Latin-1 translation table

00 01 02 03 9C 09 86 7F 97 8D 8E 0B 0C 0D 0E 0F
10 11 12 13 9D 85 08 87 18 19 92 8F 1C 1D 1E 1F
80 81 82 83 84 0A 17 1B 88 89 8A 8B 8C 05 06 07
```
(line breaks can be anywhere)

Reply:
OK, I see. I've thought that each non-comment, non-empty line is a "data set", but those can span over multiple lines - which is likely much easier to read.

So each data set ends after 256 characters now? [one data set row is likely easier to implement and _maybe_ easier to understand when looking at the file]

Then the file either needs 2 data sets (conversion is done from the first to the second and vice versa) or 4 data sets (conversion is done from 1 to 2 in the one direction and from 3 to 4 for the other direction).

if this doesn't makes sense then I've not understood something, please elaborate in this case.

Reply:
Correct. So your file should either have one or two data sets, if only one is given the other one is implicit (it is just important to note that in the format documentation in gnucobol.texi; ideally we have at least two ttbl we ship and can reference two of those with one being symmetric, the other one not.

Reply:
Yes, the old "built in" tables are most likely not symmetric, and the new EBCDIC 500 to Latin-1 is, so there will be examples of both cases.

Reply:
Almost done, just this documentation to add. In which section should I add the format description ?

Reply:
> In which section should I add the format description ?

Please add a short description on `-febcdic-table` as new paragraph at the end of "2.1.5 Configuration options". We could document the format in default.ttbl (likely a good idea in general) and generate an appendix from it, similar as it is done for other files (I could help with the generation part if necessary) - but this may be overkill; or just reference the file within the explanation "as seen in default.ttbl in the configuration directory (see `cobc --info`), ...".

When on it: please drop `-fmfcomment`, `-acucoment` and `-fsign` from 2.1.8 where they don't belong any more.

Additional it would be useful to add the new external functions as 2.3.7 (its enough to give a short blurb and the function signature); but "missing" that won't hinder the PR to get merged. People should not need to inspect libcob's source code to know about these functions and where they may be useful and how they should be called.